src/ephemeris/set_library_permissions.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+"""Tool to set permissions for all datasets of a given Galaxy Data Library"""
+
+import argparse
+import logging as log
+from bioblend import galaxy
+import sys
+from rich.progress import Progress
+from .common_parser import get_common_args
+
+# Print iterations progress
+
+
+def get_datasets(gi, library_id) -> [str]:
+    objects = gi.libraries.show_dataset(library_id=library_id, dataset_id="")
+    datasets = []
+    for index in range(len(objects)):
+        if objects[index]["type"] == "file":
+            datasets.append(objects[index]["id"])
+    if datasets == []:
+        sys.exit("No datasets in library!")
+    else:
+        return datasets
+
+
+def set_permissions(gi, library_id, role_ids, auto):
+    log.info("Your library_id is " + library_id + "\n")
+    log.info("Your roles are: %s", " ".join(role_ids))
+    datasets = get_datasets(gi, library_id)
+    total = len(datasets)
+    est = total * 3 / 60
+    # Give User time to abort
+    log.info(
+        "\nSuccess! %d datasets found. Processing can take up to %0.02f min\n",
+        total,
+        est,
+    )
+    if auto:
+        for current in range(total):
+            log.debug(
+                "Processing dataset %d of %d, ID=%s", current, total, datasets[current]
+            )
+            gi.libraries.set_dataset_permissions(
+                dataset_id=datasets[current],
+                access_in=role_ids,
+                modify_in=role_ids,
+                manage_in=role_ids,
+            )
+    else:
+        if input("Do you want to continue? (y/n) ") == "y":
+            with Progress() as progress:
+                task = progress.add_task("[green]Processing datasets...", total=total)
+                for current in range(total):
+                    log.debug(
+                        "Processing dataset %d of %d, ID=%s",
+                        current,
+                        total,
+                        datasets[current],
+                    )
+                    gi.libraries.set_dataset_permissions(
+                        dataset_id=datasets[current],
+                        access_in=role_ids,
+                        modify_in=role_ids,
+                        manage_in=role_ids,
+                    )
+                    progress.update(task, advance=1)
+        else:
+            log.info("Operation cancelled by user. No changes were applied.\n")
+
+
+def _parser():
+    """Constructs the parser object"""
+    parent = get_common_args()
+    parser = argparse.ArgumentParser(
+        parents=[parent], description="Populate the Galaxy data library with data."
+    )
+    parser.add_argument("--library", help="Specify the data library ID")
```

Comment:
```suggestion
    parser.add_argument("library", help="Specify the data library ID")
```

since this is mandatory, maybe makes sense to just require it outright